### PR TITLE
Update UI glossary

### PR DESCRIPTION
**Pull request by GerardasB**
This PR updates UI glossary.

### CODE REVIEW

File: docs/learning/ui/UIGlossary.md [hunk 1]
```diff
@@ -6,20 +6,16 @@ tableRowAnchors: true
...
-|**Tool&nbsp;Settings**|Properties that modify the behavior and outcome of the active tool. Typically docked as a bar at the top of the application window.
-|**View&nbsp;Navigation&nbsp;Tools**|These tools are located in the right side of the content area. These tools are used to navigate and modify the view parameter such as rotation and zoom level.
-|**Widget**|A collection of UI components tied to a particular section that allows the user to view and/or modify data relevant to their current context.
+|**Backstage**|The main navigation menu of the application used to switch between frontstages, open overlays, or launch custom commands. You can use [BackstageComposer]($appui-react) to set up a backstage that displays an overlay menu along the left edge of the application.|
+|**Content&nbsp;Control**|Control that is displayed in a content layout. You can use [ContentControl]($appui-react) to specify a custom React component.|
+|**Content&nbsp;Group**|Defines which content controls are rendered and how in the content area of a frontstage. You can use [ContentGroup]($appui-react) to describe a collection of content controls and their content layout.|
```
Comment (ben-polinsky) [inline docs/learning/ui/UIGlossary.md:11]: "Defines which content controls are rendered and how in the content area of a frontstage" is a bit awkward. 

Is the "which" necessary?
> Defines how content controls are rendered in the content area of a frontstage.

File: docs/learning/ui/UIGlossary.md [hunk 2]
```diff
@@ -6,20 +6,16 @@ tableRowAnchors: true
...
+|**Stage&nbsp;Panel&nbsp;Section**|A dedicated area within a stage panel that can contain multiple widgets.|
+|**Tool&nbsp;Settings**|A dedicated area in the user interface that displays properties or information related to the active tool.|
+|**UI&nbsp;Items&nbsp;Provider**|A mechanism for providing UI elements such as widgets, backstage, toolbar or status bar items to the application. You can register a [UiItemsProvider]($appui-react) to provide additional items.|
+|**Widget**|An interactive UI element for a custom content of an application that allows the user to view and/or modify data relevant to their current context. The content of the [Widget]($appui-react) is just a React component, but additional meta-data such as label or icon can be provided to customize or initialize the widget. Multiple widgets can be grouped and displayed in one of widget containers.<br/>**Docked** - when a widget is docked to one of stage panel sections on the side of the page.<br/>**Floating** - when a widget is displayed in a dialog like component of the page.<br/>**Popout** - when a widget is displayed in a separate popup window.|
```
Comment (ben-polinsky) [inline docs/learning/ui/UIGlossary.md:21]: "and displayed in one of widget containers" should be "and displayed in one of *the* widget containers"? Or maybe just "and displayed in widget containers"?

Comment (ben-polinsky) [reply docs/learning/ui/UIGlossary.md:21]: Similar issue here "when a widget is docked to one of stage panel sections"

How about "When a widget is docked to a stage panel section..."

Comment (ben-polinsky) [reply docs/learning/ui/UIGlossary.md:21]: "dialog like" should be "dialog-like"

File: docs/learning/ui/UIGlossary.md (hunk 1 above)
Comment (ben-polinsky) [reply docs/learning/ui/UIGlossary.md:11]: Eh, circling back I can live with this phrasing. Unless you like my suggestion better.

Comment (GerardasB) [reply docs/learning/ui/UIGlossary.md:11]: Updated.

File: docs/learning/ui/UIGlossary.md (hunk 2 above)
Comment (GerardasB) [reply docs/learning/ui/UIGlossary.md]: Fixed.

**Review by ben-polinsky — APPROVED** [review]

**Review by raplemie — APPROVED** [review]
LGTM

**Comment by raplemie** [comment]
@ben-polinsky I added visuals and reorganized the glossary, so I'd like you to review again when you can before I merge.

File: docs/learning/ui/UIGlossary.md [hunk 3]
```diff
@@ -5,22 +5,68 @@ layout: custom-ui-feedback.handlebars
...
+
+![Tool settings visual](images/tool-settings.svg)
+
+## UI&nbsp;Items&nbsp;Provider
+
+A mechanism for providing UI elements such as widgets, backstage, toolbar or status bar items to the application. You can register a [UiItemsProvider]($appui-react) to provide additional items.
+
+## Widget
+
+An interactive UI element for a custom content of an application that allows the user to view and/or modify data relevant to their current context. The content of the [Widget]($appui-react) is just a React component, but additional meta-data such as label or icon can be provided to customize or initialize the widget. Multiple widgets can be grouped and displayed in one of the widget containers.
```
Comment (ben-polinsky) [inline docs/learning/ui/UIGlossary.md:64]:
```suggestion
An interactive UI element for a custom content of an application that allows the user to view and/or modify data relevant to their current context. The content of the [Widget]($appui-react) is just a React component, but additional metadata such as label or icon can be provided to customize or initialize the widget. Multiple widgets can be grouped and displayed in one of the widget containers.
```

**Comment by ben-polinsky** [comment]
> @ben-polinsky I added visuals and reorganized the glossary, so I'd like you to review again when you can before I merge.

Aside from meta-data -> metadata, there are a few types which are not linked such as BackstageComposer, NavigationAidControl, Viewport, FrontstageProvider, etc... Intentional or no?

File: docs/learning/ui/UIGlossary.md (hunk 3 above)
Comment (raplemie) [reply docs/learning/ui/UIGlossary.md]: Updated

**Review by ben-polinsky — APPROVED** [review]

**Comment by ben-polinsky** [comment]
@mergifyio backport

**Pull request merged**